flink-connectors/flink-connector-elasticsearch7/src/main/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchSinkBuilder.java

```diff
@@ -0,0 +1,276 @@
+        this.bulkFlushMaxActions = numMaxActions;
+        return this;
+    }
+
+    /**
+     * Sets the maximum size of buffered actions, in mb, per bulk request. You can pass -1 to
+     * disable it.
+     *
+     * @param maxSizeMb the maximum size of buffered actions, in mb.
+     * @return {@link ElasticsearchSinkBuilder}
+     */
+    public ElasticsearchSinkBuilder<IN> setBulkFlushMaxSizeMb(int maxSizeMb) {
+        checkState(
+                maxSizeMb == -1 || maxSizeMb > 0,
+                "Max size of buffered actions must be larger than 0.");
+        this.bulkFlushMaxMb = maxSizeMb;
+        return this;
+    }
+
+    /**
+     * Sets the bulk flush interval, in milliseconds. You can pass -1 to disable it.
+     *
+     * @param intervalMillis the bulk flush interval, in milliseconds.
+     * @return {@link ElasticsearchSinkBuilder}
+     */
+    public ElasticsearchSinkBuilder<IN> setBulkFlushInterval(long intervalMillis) {
+        checkState(
+                intervalMillis == -1 || intervalMillis >= 0,
+                "Interval (in milliseconds) between each flush must be larger than "
+                        + "or equal to 0.");
+        this.bulkFlushInterval = intervalMillis;
+        return this;
+    }
+
+    /**
+     * Sets the type of back of to use when flushing bulk requests.
+     *
+     * @param flushBackoffType the backoff type to use.
+     * @return {@link ElasticsearchSinkBuilder}
+     */
+    public ElasticsearchSinkBuilder<IN> setBulkFlushBackoffType(
+            ElasticsearchSinkBase.FlushBackoffType flushBackoffType) {
+        this.bulkFlushBackoffType = checkNotNull(flushBackoffType);
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of retries for a backoff attempt when flushing bulk requests.
+     *
+     * @param maxRetries the maximum number of retries for a backoff attempt when flushing bulk
+     *     requests
+     * @return {@link ElasticsearchSinkBuilder}
+     */
+    public ElasticsearchSinkBuilder<IN> setBulkFlushBackoffRetries(int maxRetries) {
+        checkState(maxRetries > 0, "Max number of backoff attempts must be larger than 0.");
+        this.bulkFlushBackoffRetries = maxRetries;
+        return this;
+    }
+
+    /**
+     * Sets the amount of delay between each backoff attempt when flushing bulk requests, in
+     * milliseconds.
+     *
+     * @param delayMillis the amount of delay between each backoff attempt when flushing bulk
+     *     requests, in milliseconds.
+     * @return {@link ElasticsearchSinkBuilder}
+     */
+    public ElasticsearchSinkBuilder<IN> setBulkFlushBackoffDelay(long delayMillis) {
+        checkState(
+                delayMillis >= 0,
+                "Delay (in milliseconds) between each backoff attempt must be larger "
+                        + "than or equal to 0.");
+        this.bulkFlushBackOffDelay = delayMillis;
+        return this;
+    }
```

Comment:
This feels like we will add in one or another way to all the batchy sink builders :/ (so all of AsyncSink's implementations). I think this is the perfect time to extract some base builders (or maybe when adopting AsyncSink?)

Reply:
Definitely 👍